src/twisted/internet/test/test_inlinecb.py

```diff
@@ -83,6 +96,26 @@ def _genBasics():
 
         return _genBasics().addCallback(self.assertEqual, "WOOSH")
 
+    def testBasicsAsync(self):
+        """
+        Test that a inlineCallbacks that yields coroutine works.
+        Alse tests functions that throw exceptions.
+        """
+
+        @inlineCallbacks
+        def _genBasics():
+            x = yield getValueViaCoro("hi")
+
+            self.assertEqual(x, "hi")
+
+            try:
+                yield getDivisionFailureCoro("OMG")
+            except ZeroDivisionError as e:
+                self.assertEqual(str(e), "OMG")
+                returnValue("WOOSH")
```

Comment:
Why not use the standard `return` statement ?

```suggestion
                return "WOOSH"
```

Reply:
This tests that `returnValue` still works. Once we actually remove the feature we can simplify the tests too.

Reply:
My worry is that returValue now raises a deprecation warning and these new tests are not asserting or clearing that warning.

This is not a big issue now... but it adds some tech debt.

In an ideal case, we will run each test while capturing any warning and failing if there are unhandled / un-asserted warnings.

Unfortunately, `trial` has no support for automatic catching of the warnings that are generated during a test run... and this is why the test are not yet failing.

-------

I understand that the current tests have a wider coverage ... so we can keep this code.

-----

I was thinking that since `returnValue` is now a thing of the past, we can stop worrying about it and don't bother too much with covering `returnValue` in new tests.

If `returnValue` don't work with coroutine, that should be fine... as the original code written using `returnValue` is old code from the time when coroutines were not supported.